### PR TITLE
feat: add menu component - INNO-1603

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Add beta menu component according to [specifications](https://webgate.ec.europa.eu/CITnet/confluence/display/NEXTEUROPA/Menu_ec+Design+Specifications+-+WIP)

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 30cdf37aaa8632da4714dc90b552e904e8b941a8

https://deploy-preview-946--europa-component-library.netlify.com

File: src/systems/ec/implementations/vanilla/packages/ec-preset-legacy-website/package.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    "build": "rimraf build && run-p build:* -ln",
+    "clean": "rimraf build dist",
+    "dist": "npm run clean && cross-env NODE_ENV=production run-p build:* -ln",
+    "watch:scripts": "chokidar \"../../../../generic/**/*.js\" \"../../**/*.js\" -i \"/node_modules|build|dist|config\\.js/\" -c \"npm run build:scripts\"",
+    "watch:styles": "chokidar \"../../../../generic/**/*.scss\" \"../../**/*.scss\" -i \"/node_modules|build|dist/\" -c \"npm run build:styles\"",
+    "watch": "npm-run-all --parallel watch:* -ln",
+    "prepublish": "npm run dist"
+  },
+  "dependencies": {
+    "@ecl/ec-component-menu-legacy": "^2.7.0"
```
Comment (yhuard) [inline src/systems/ec/implementations/vanilla/packages/ec-preset-legacy-website/package.json]: should be a `devDependencies` (to be consistent with `@ecl/ec-preset-dev` and others)

**Review by yhuard — DISMISSED** [review]

File: src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.scss [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+  .ecl-menu[aria-expanded='true'] .ecl-menu__hamburger--bar1 {
+    transform: rotate(-45deg) translate(-9px, 6px);
+  }
+
+  .ecl-menu[aria-expanded='true'] .ecl-menu__hamburger--bar2 {
+    opacity: 0;
+  }
+
+  .ecl-menu[aria-expanded='true'] .ecl-menu__hamburger--bar3 {
+    transform: rotate(45deg) translate(-8px, -8px);
```
Comment (degliwe) [inline src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.scss]:
```suggestion
    transform: rotate(45deg) translate(-9px, -8px);
```

File: src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.scss [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+  .ecl-menu__hamburger--bar3 {
+    background-color: $toggle-color;
+    height: 5px;
+    margin: 6px 0;
+    transition: 0.4s;
+    width: 35px;
+  }
+
+  .ecl-menu[aria-expanded='true'] .ecl-menu__hamburger--bar1 {
+    transform: rotate(-45deg) translate(-9px, 6px);
```
Comment (degliwe) [inline src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.scss]:
```suggestion
    transform: rotate(-45deg) translate(-8px, 6px);
```

File: src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.scss [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+      &::before {
+        background-color: $subitem-background-desktop;
+        display: block;
+      }
+    }
+
+    .ecl-menu__subitem::before {
+      display: none;
+    }
+
```
Comment (degliwe) [inline src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.scss:364]:
```suggestion
  .ecl-menu__subitem {
    padding-right: $ecl-spacing-m ;
    box-sizing: border-box;
  }
```

**Review by degliwe — CHANGES_REQUESTED** [review]

File: src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    while (el) {
+      if (el.matches(selector)) {
+        return el;
+      }
+      el = el.parentElement;
+    }
+    return null;
+  };
+
+class Menu {
```
Comment (yhuard) [inline src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.js]:
```suggestion
export class MenuLegacy {
```

File: src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+        'aria-hidden',
+        menuMega.getAttribute('aria-hidden') === 'false' ? 'true' : 'false'
+      );
+    }
+
+    return this;
+  }
+}
+
+export default Menu;
```
Comment (yhuard) [inline src/systems/ec/implementations/vanilla/packages/ec-component-menu-legacy/ec-component-menu-legacy.js]:
```suggestion
export default MenuLegacy;
```

File: src/systems/eu/implementations/vanilla/packages/eu-component-menu-legacy/eu-component-menu-legacy.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    while (el) {
+      if (el.matches(selector)) {
+        return el;
+      }
+      el = el.parentElement;
+    }
+    return null;
+  };
+
+class Menu {
```
Comment (yhuard) [inline src/systems/eu/implementations/vanilla/packages/eu-component-menu-legacy/eu-component-menu-legacy.js]:
```suggestion
export class MenuLegacy {
```

File: src/systems/eu/implementations/vanilla/packages/eu-component-menu-legacy/eu-component-menu-legacy.js [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+        'aria-hidden',
+        menuMega.getAttribute('aria-hidden') === 'false' ? 'true' : 'false'
+      );
+    }
+
+    return this;
+  }
+}
+
+export default Menu;
```
Comment (yhuard) [inline src/systems/eu/implementations/vanilla/packages/eu-component-menu-legacy/eu-component-menu-legacy.js]:
```suggestion
export default MenuLegacy;
```

**Review by yhuard — CHANGES_REQUESTED** [review]

**Review by yhuard — APPROVED** [review]

**Review by degliwe — APPROVED** [review]

**Pull request merged**